tests/conftest.py

```diff
@@ -25,8 +25,12 @@ def set_mock(obj, attr, mock):
 def tracking_uri_mock(tmpdir, request):
     try:
         if 'notrackingurimock' not in request.keywords:
-            mlflow.set_tracking_uri(path_to_local_sqlite_uri(
-                os.path.join(tmpdir.strpath, 'mlruns')))
+            tracking_uri = path_to_local_sqlite_uri(
+                os.path.join(tmpdir.strpath, 'mlruns'))
+            mlflow.set_tracking_uri(tracking_uri)
+            os.environ["MLFLOW_TRACKING_URI"] = tracking_uri
```

Comment:
This is what was causing tests that launch subprocesses to fail (e.g. the CLI tests that we marked with `pytest.mark.notrackingurimock`). The specific problem is that we'd set the tracking URI, but not set the MLFLOW_TRACKING_URI environment variable, so subprocesses would not run with our mock tracking URI (& instead would actually log to whatever the user's default tracking URI was, usually ./mlruns within the current directory).